ts/Series/Treegraph/TreegraphLayout.ts

```diff
@@ -0,0 +1,430 @@
+/* *
+ *
+ *  (c) 2010-2022 Pawel Lysy Grzegorz Blachlinski
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+import type TreegraphSeries from './TreegraphSeries';
+import H from '../../Core/Globals.js';
+import U from '../../Core/Utilities.js';
+const { extend } = U;
+import TreegraphNode from './TreegraphNode.js';
+
+declare global {
+    namespace Highcharts {
+        class TreegraphLayout {
+            public constructor();
+            public calculatePositions(series: TreegraphSeries): void;
+            public resetValues(nodes: TreegraphNode.Node[]): void;
+            public calculateRelativeX(
+                root: TreegraphNode.Node,
+                index: number
+            ): void;
+            public beforeLayout(nodes: TreegraphNode.Node[]): void;
+            public afterLayout(nodes: TreegraphNode.Node[]): void;
+            public firstWalk(node: TreegraphNode.Node): void;
+            public secondWalk(node: TreegraphNode.Node, modSum: number): void;
+            public executeShifts(node: TreegraphNode.Node): void;
+            public apportion(
+                node: TreegraphNode.Node,
+                defaultAncestor: TreegraphNode.Node
+            ): TreegraphNode.Node;
+            public setArea(x: number, y: number, w: number, h: number): void;
+            public moveSubtree(
+                ancestor: TreegraphNode.Node,
+                node: TreegraphNode.Node,
+                shift: number
+            ): void;
+            public box: Record<string, number>;
+            public static createDummyNode(
+                parentNode: TreegraphNode.Node,
+                childNode: TreegraphNode.Node,
+                gap: number,
+                index: number
+            ): TreegraphNode.Node;
+        }
+        let treeLayouts: Record<string, typeof TreegraphLayout>;
+    }
+}
+
+H.treeLayouts = {
+    Walker: function (): void {}
+} as any;
```

Comment:
I'm curious if we need to pollute the Highcharts namespace. I know it's done that way in Networkgraph, but perhaps it's unnecessary there too? I'm not saying you should change the Networgraph series too

Reply:
Could we simply export the layout?